src/main/java/co/cask/gcp/datastore/util/DatastorePropertyUtil.java

```diff
@@ -0,0 +1,149 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.gcp.datastore.util;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.hydrator.common.KeyValueListParser;
+import com.google.cloud.datastore.PathElement;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.datastore.v1.client.DatastoreHelper;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Utility class that provides method for validation, parsing and conversion of config properties.
+ */
+public class DatastorePropertyUtil {
+
+  private static final KeyValueListParser KV_PARSER = new KeyValueListParser(";", "\\|");
+  private static final Pattern KEY_PATTERN = Pattern.compile("\\s*key\\s*\\((.*)\\)\\s*", Pattern.CASE_INSENSITIVE);
+  public static final String DEFAULT_NAMESPACE = "";
+
+  /**
+   * Checks if given namespace value is not null or empty,
+   * otherwise returns {@link DatastorePropertyUtil#DEFAULT_NAMESPACE}.
+   *
+   * @param namespace namespace value
+   * @return valid namespace value
+   */
+  public static String getNamespace(String namespace) {
+    return Strings.isNullOrEmpty(namespace) ? DEFAULT_NAMESPACE : namespace;
+  }
+
+  /**
+   * Parser JSON schema into CDAP schema object.
+   *
+   * @param schema schema in JSON format
+   * @return CDAP schema
+   */
+  public static Schema parseSchema(String schema) {
+    try {
+      return Schema.parseJson(schema);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Unable to parse output schema: " + schema, e);
+    }
+  }
+
+  /**
+   * Parsers given string value into map of strings.
+   * If given value is null or empty, returns empty map.
+   *
+   * @param value string value
+   * @return map with parsed string value
+   */
+  public static Map<String, String> parseKeyValuePairs(String value) {
+    if (Strings.isNullOrEmpty(value)) {
+      return Collections.emptyMap();
+    }
+    return StreamSupport.stream(KV_PARSER.parse(value).spliterator(), false)
+      .collect(Collectors.toMap(
+        KeyValue::getKey,
+        KeyValue::getValue,
+        (o, n) -> n,
+        LinkedHashMap::new));
+  }
+
+  /**
+   * Parses given key literal string value into list of path elements
+   * according to {@link DatastorePropertyUtil#KEY_PATTERN}.
+   *
+   * @param keyLiteral key literal
+   * @return list of path elements
+   */
+  public static List<PathElement> parseKeyLiteral(String keyLiteral) {
```

Comment:
Isn't this format something Datastore uses? None of their classes have anything for this?